File: 3515-FindIfDigitGameCanBeWon/3515-FindIfDigitGameCanBeWon.java
// Last updated: 11/07/2026, 12:31:47
class Solution {
    public boolean canAliceWin(int[] nums){
        int single = 0, doubleDigit = 0;

        for (int n : nums) {
            if (n < 10) single += n;
            else doubleDigit += n;
        }

        int total = single + doubleDigit;
        boolean aliceSingle = single > total - single;

        boolean aliceDouble = doubleDigit > total - doubleDigit;
        return aliceSingle || aliceDouble;
    }
}

File: 3515-FindIfDigitGameCanBeWon/3515-FindIfDigitGameCanBeWon.cpp
// Last updated: 11/07/2026, 12:31:40
class Solution {
public:
    bool canAliceWin(vector<int>& nums) {
        int singleDigit = 0, doubleDigit = 0;
        for (int x:nums) {
            if(x<10) singleDigit+=x;
            else doubleDigit += x;
        }
        return doubleDigit!=singleDigit ? true:false;
    }
};